### ISSUE TITLE
Devices often become unavailable and need restart of HA

### ISSUE THREAD

**Issue by lukejagger** [albertogeniola/meross-homeassistant#125]
Any network blip and it seems the Meross devices become unavailable in HA, when they're all still up in the Meross app.  It then requires a restart of HA to get them updating again properly.  This has been an issue for quite a while now.

**Comment by albertogeniola** [comment]
Hi @lukejagger , can you provide some error logs about the errors you are facing?

**Comment by cree8** [comment]
This happens to me **all the time ** as well. I figured it was a known issue. I have the following errors

`2020-09-13 09:43:35 INFO (Thread-8) [meross_iot.manager] Disconnection detected. Reason: 1
2020-09-13 09:43:35 WARNING (Thread-8) [meross_iot.manager] Client has been disconnected, however auto_reconnect flag is set. Won't stop the looping thread, as it will retry to connect.
2020-09-13 09:43:35 WARNING (MainThread) [custom_components.meross_cloud.cover] Device Smart Garage Door Opener (msg100) - Main channel reported online event.

2020-09-13 09:44:27 WARNING (Thread-8) [meross_iot.manager] Subscribed to topics, updating state for already known devices...

2020-09-13 09:44:32 ERROR (MainThread) [meross_iot.manager] Timeout occurred while waiting a response for message b'{"header": {"from": "/app/384830-bce2cccc22b603fd12699315a1f1a424/subscribe", "messageId": "fdc5c663ef9aec42b978eb562695ed62", "method": "GET", "namespace": "Appliance.System.All", "payloadVersion": 1, "sign": "2a76dba8cf31d0fca90953b9e45a096e", "timestamp": 1600008268}, "payload": {}}' sent to device uuid 1909036899479690800248e1e9513b1e. Timeout was: 5 seconds
2020-09-13 09:44:32 ERROR (MainThread) [homeassistant] Error doing job: _GatheringFuture exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/meross_iot/manager.py", line 536, in _async_send_and_wait_ack
    return await asyncio.wait_for(future, timeout, loop=self._loop)
  File "/usr/local/lib/python3.8/asyncio/tasks.py", line 490, in wait_for
    raise exceptions.TimeoutError()
asyncio.exceptions.TimeoutError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/meross_iot/controller/mixins/system.py", line 24, in async_update
    result = await self._execute_command(method="GET", namespace=Namespace.SYSTEM_ALL, payload={})
  File "/usr/local/lib/python3.8/site-packages/meross_iot/controller/device.py", line 196, in _execute_command
    return await self._manager.async_execute_cmd(destination_device_uuid=self.uuid,
  File "/usr/local/lib/python3.8/site-packages/meross_iot/manager.py", line 527, in async_execute_cmd
    response = await self._async_send_and_wait_ack(future=fut,
  File "/usr/local/lib/python3.8/site-packages/meross_iot/manager.py", line 540, in _async_send_and_wait_ack
    raise CommandTimeoutError()
meross_iot.model.exception.CommandTimeoutError`

**Comment by lenny1972** [comment]
I can confirm the same problem with some meross devices I bought (smart plug mss310r) and integrate in Home assistant using your integration. The strange thing is that for each and every device I can read the volt, current and consumption info but the on/off switches are not enabled (are in grey). I've attach a picture to show the problem (see the plug named "Presa Meross #2 (APA0)"). I've also noticed that in order to restore the on/off switches of the plugs in HA I have to switch it on (or off depending on their starting state) either from the physical button on the plug itself or by using the official Meross App on my android phone. After that the on/off switches in the Home Assistant Lovelace panel becomes blue and start to work correctly. Another option will be to restart HA as cree8 suggested.
![meross plug problem](https://user-images.githubusercontent.com/13016843/93972084-ea2a6480-fd71-11ea-9f4d-2a06c7791e5f.png)

**Comment by silviudc** [comment]
I am now experiencing the same issue many disconnects during the day and must restart HA to get all meross devices back.

```
2020-10-05 13:52:17 INFO (Thread-6) [meross_iot.manager] Disconnection detected. Reason: 5
2020-10-05 13:52:17 WARNING (Thread-6) [meross_iot.manager] Client has been disconnected, however auto_reconnect flag is set. Won't stop the looping thread, as it will retry to connect.
```

Any idea on a fix, happens a lot recently

**Comment by albertogeniola** [comment]
Hi @cree8 ,
can you provide more details about your devices? HW version and FW version?

**Comment by Wapiya** [comment]
I also have this kind of problems since I am using HA and Meross, erverything runs perfect for an unspecified while then either the swiitch or the watt sensor becomes unavailable. When in this state the device are still usable with the meross app. A restart of HA always solves the problem immidiatly, so I did the following work around:

    _trigger_heiz_err_power:
      friendly_name: "Trigger: Meross Watt-Sensors"
      value_template: >-
        {% if is_state('sensor.wozi_oben_mss310_power_sensor_w_0', 'unavailable') %}
          offline
.....list all meross devices here....
        {% else %}
          online
        {% endif %}

    _trigger_heiz_err_switch:
      friendly_name: "Trigger: Meross Switches"
      value_template: >-
        {% if is_state("switch.wozi_oben_mss310_main_channel","o*") %}
          offline
.....list all meross devices here....
        {% else %}
          online
        {% endif %}__

Afterwards I set an auotmation which send me a message that HA has to be restartet to bring up meross again - so at least I can use them now continously. The automation disables it self (to avoid loops during internet loss e.g.) and aonther rule is enabling it after a while again :-D 

I am using nine mss310 one mss425e and two mss710 - If I can provide any logfile of interrest, please contact me. I also can take part in testing the offline version, if the can run side by side

**Comment by silviudc** [comment]
Happening again in 0.117.4 and it weirdly says X device reported an online event but none are online actually.
```
2020-11-05 07:58:22 WARNING (MainThread) [custom_components.meross_cloud.sensor] Device Garden lights (mss310) - power sensor (W, 0) reported online event.
2020-11-05 07:58:22 WARNING (MainThread) [custom_components.meross_cloud.sensor] Device Garden lights (mss310) - power sensor (A, 0) reported online event.
2020-11-05 07:58:22 WARNING (MainThread) [custom_components.meross_cloud.sensor] Device Garden lights (mss310) - power sensor (V, 0) reported online event.
2020-11-05 07:58:22 WARNING (MainThread) [custom_components.meross_cloud.switch] Device Garden lights (mss310) - Main channel reported online event.
2020-11-05 08:00:22 WARNING (Thread-6) [meross_iot.manager] Client has been disconnected, however auto_reconnect flag is set. Won't stop the looping thread, as it will retry to connect.
```

**Comment by albertogeniola** [comment]
Hi guys, version v1.0.4 should mitigate it. Can you please test it and report if that helps?

**Comment by silviudc** [comment]
Still happening for me and also the log full of these:
```
2020-12-02 17:07:47 WARNING (MainThread) [meross_iot.manager] Rate limit reached (RateLimitResult.GlobalLimitReached): api call will be delayed by 0.5 seconds
2020-12-02 17:07:48 WARNING (MainThread) [meross_iot.manager] Rate limit reached (RateLimitResult.GlobalLimitReached): api call will be delayed by 1.0 seconds
2020-12-02 17:07:49 WARNING (MainThread) [meross_iot.manager] Rate limit reached (RateLimitResult.GlobalLimitReached): api call will be delayed by 0.5 seconds
2020-12-02 17:07:50 WARNING (MainThread) [meross_iot.manager] Rate limit reached (RateLimitResult.GlobalLimitReached): api call will be delayed by 1.0 seconds
2020-12-02 17:07:51 WARNING (MainThread) [meross_iot.manager] Rate limit reached (RateLimitResult.GlobalLimitReached): api call will be delayed by 0.5 seconds
2020-12-02 17:07:51 WARNING (MainThread) [meross_iot.manager] Rate limit reached (RateLimitResult.GlobalLimitReached): api call will be delayed by 1.0 seconds
2020-12-02 17:07:52 WARNING (MainThread) [meross_iot.manager] Rate limit reached (RateLimitResult.GlobalLimitReached): api call will be delayed by 0.5 seconds
2020-12-02 17:07:53 WARNING (MainThread) [meross_iot.manager] Rate limit reached (RateLimitResult.GlobalLimitReached): api call will be delayed by 1.0 seconds
2020-12-02 17:07:54 WARNING (MainThread) [meross_iot.manager] Rate limit reached (RateLimitResult.GlobalLimitReached): api call will be delayed by 0.5 seconds
2020-12-02 17:07:54 WARNING (MainThread) [meross_iot.manager] Rate limit reached (RateLimitResult.GlobalLimitReached): api call will be delayed by 1.0 seconds
```

**Comment by Wapiya** [comment]
Hello Alberto,

I got the issue that they got unavailable twice since the 
update......but it stays longer. Do you need any logfiles for debugging?

regards Steffen

Am 03.12.2020 00:08, schrieb Alberto Geniola:

> Closed #125 [1].
> 
> --
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub [2], or unsubscribe 
> [3].


Links:
------
[1] https://github.com/albertogeniola/meross-homeassistant/issues/125
[2] 
https://github.com/albertogeniola/meross-homeassistant/issues/125#event-4064772253
[3]
